crates/controller/src/areaselect.rs

```diff
@@ -66,6 +79,10 @@ fn select_in_area(
         let event_frustum = screen_frustum.rect(in_event.rect());
         let entities: Vec<Entity> = candidates
             .iter()
+            .filter(|(_, &object_type, _)| {
+                in_event.filter_object_type().is_none()
```

Comment:
I would slightly prefer something like this: `in_event.filter_object_type().map_or(true, |filter| filter == object_type)`